### PR TITLE
connect components using watchable

### PR DESCRIPTION
**Pull request by arkodg**
Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

File: internal/cmd/server.go [hunk 1]
```diff
@@ -49,19 +67,82 @@ func server() error {
...
+	// https://github.com/envoyproxy/gateway/issues/43
+	ctx := ctrl.SetupSignalHandler()
 
-	k8sTable := new(provider.ResourceTable)
-	if err := provider.Start(cfg, k8sTable); err != nil {
+	pResources := new(message.ProviderResources)
+	// Start the Provider Service
+	// It fetches the resources from the configured provider type
+	// and publishes it
+	providerSvc := &providersvc.Service{
```
Comment (danehans) [inline internal/cmd/server.go:90]: `providersvc.Service` studders. Should we make `Service` as part of the component pkg so we have `provider.Service`?

File: internal/cmd/server.go [hunk 2]
```diff
@@ -49,19 +67,82 @@ func server() error {
...
+	if err := providerSvc.Start(ctx); err != nil {
+		return err
+	}
+
+	xdsIR := new(message.XdsIR)
+	infraIR := new(message.InfraIR)
+	// Start the GatewayAPI Translator Service
+	// It subscribes to the provider resources, translates it to xDS IR
+	// and infra IR resources and publishes them.
+	gwSvc := &gatewayapisvc.Service{
```
Comment (danehans) [inline internal/cmd/server.go:103]: Same comment as https://github.com/envoyproxy/gateway/pull/197/files#r942653954.

File: internal/cmd/server.go [hunk 3]
```diff
@@ -49,19 +67,82 @@ func server() error {
...
+		InfraIR:           infraIR,
+	}
+	if err := gwSvc.Start(ctx); err != nil {
+		return err
+	}
+
+	xResources := new(message.XdsResources)
+	// Start the Xds Translator Service
+	// It subscribes to the xdsIR, translates it into xds Resources and publishes it.
+	xdsTranslatorSvc := &xdstranslatorsvc.Service{
```
Comment (danehans) [inline internal/cmd/server.go:116]: Same comment as https://github.com/envoyproxy/gateway/pull/197/files#r942653954.

File: internal/cmd/server.go [hunk 4]
```diff
@@ -49,19 +67,82 @@ func server() error {
...
+		XdsResources: xResources,
+	}
+	if err := xdsTranslatorSvc.Start(ctx); err != nil {
+		return err
+	}
+
+	// Start the Infra Manager Service
+	// It subscribes to the infraIR, translates it into Envoy Proxy infrastructure
+	// resources such as K8s deployment and services.
+	infraSvc := &infrasvc.Service{
```
Comment (danehans) [inline internal/cmd/server.go:128]: Same comment as https://github.com/envoyproxy/gateway/pull/197/files#r942653954.

File: internal/cmd/server.go [hunk 5]
```diff
@@ -49,19 +67,82 @@ func server() error {
...
+		InfraIR: infraIR,
+	}
+	if err := infraSvc.Start(ctx); err != nil {
+		return err
+	}
+
+	// Start the xDS Server
+	// It subscribes to the xds Resources and configures the remote Envoy Proxy
+	// via the xDS Protocol
+	xdsServerSvc := &xdsserversvc.Service{
```
Comment (danehans) [inline internal/cmd/server.go:139]: Same comment as https://github.com/envoyproxy/gateway/pull/197/files#r942653954.

File: internal/gatewayapi/service/service.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+	ProviderResources *message.ProviderResources
+	XdsIR             *message.XdsIR
+	InfraIR           *message.InfraIR
+}
+
+func (s *Service) Name() string {
+	return "gateway-api"
+}
+
+// Start starts the GatewayAPI service
```
Comment (danehans) [inline internal/gatewayapi/service/service.go:24]: s/GatewayAPI service/GatewayAPI Translator service/

File: internal/gatewayapi/service/service.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+func (s *Service) Start(ctx context.Context) error {
+	s.Logger = s.Logger.WithValues("service", s.Name())
+	// Wait until provider resources have been initialized during startup
+	s.ProviderResources.Initialized.Wait()
+	go s.subscribeAndTranslate(ctx)
+
+	<-ctx.Done()
```
Comment (danehans) [inline internal/gatewayapi/service/service.go:31]: Should we also handle `ctx.Err()`?

File: internal/infrastructure/service/service.go [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+type Service struct {
+	config.Server
+	InfraIR *message.InfraIR
+}
+
+func (s *Service) Name() string {
+	return "infrastructure"
+}
+
+// Start starts the GatewayAPI service
```
Comment (danehans) [inline internal/infrastructure/service/service.go:19]: s/GatewayAPI/Infra Manager/

File: internal/infrastructure/service/service.go [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+func (s *Service) subscribeAndTranslate(ctx context.Context) {
+	// Subscribe to resources
+	irCh := s.InfraIR.Subscribe(ctx)
+	for ctx.Err() == nil {
+		// Receive subscribed resource notifications
+		<-irCh
+		// s.InfraIR.Get()
+		// TODO: Provision infra
+		// infrastructure.Translate(ctx, ir)
```
Comment (danehans) [inline internal/infrastructure/service/service.go:37]: xref: https://github.com/envoyproxy/gateway/pull/188

File: internal/provider/kubernetes/gateway.go [hunk 10]
```diff
@@ -30,19 +30,19 @@ type gatewayReconciler struct {
...
-	resourceTable  *ResourceTable
+	resources      *message.ProviderResources
 }
 
 // newGatewayController creates a gateway controller. The controller will watch for
 // Gateway objects across all namespaces and reconcile those that match the configured
 // gatewayclass controller name.
-func newGatewayController(mgr manager.Manager, cfg *config.Server, resourceTable *ResourceTable) error {
-	resourceTable.Initialized.Add(1)
+func newGatewayController(name string, mgr manager.Manager, logger logr.Logger, resources *message.ProviderResources) error {
```
Comment (danehans) [inline internal/provider/kubernetes/gateway.go:39]: Instead of passing in `logger logr.Logger`, consider using a context and then calling `logr.FromContext(ctx)`, [xref](https://github.com/envoyproxy/gateway/blob/main/internal/infrastructure/translator.go#L14-L24).

File: internal/provider/kubernetes/gateway.go [hunk 11]
```diff
@@ -30,19 +30,19 @@ type gatewayReconciler struct {
 	log             logr.Logger
 
 	initializeOnce sync.Once
-	resourceTable  *ResourceTable
+	resources      *message.ProviderResources
```
Comment (danehans) [inline internal/provider/kubernetes/gateway.go:33]: Note that https://github.com/envoyproxy/gateway/pull/196 adds `ResourceTable` coverage for kube provider integration tests. These tests will need to be updated for CI to pass.

File: internal/provider/kubernetes/kubernetes.go [hunk 12]
```diff
@@ -3,46 +3,34 @@ package kubernetes
...
-	client        client.Client
-	manager       manager.Manager
-	resourceTable *ResourceTable
+	client  client.Client
+	manager manager.Manager
 }
 
 // New creates a new Provider from the provided EnvoyGateway.
-func New(cfg *rest.Config, svr *config.Server, resourceTable *ResourceTable) (*Provider, error) {
+func New(controllerName string, logger logr.Logger, resources *message.ProviderResources) (*Provider, error) {
```
Comment (danehans) [inline internal/provider/kubernetes/kubernetes.go:25]: Same comment as https://github.com/envoyproxy/gateway/pull/197#discussion_r942693010.

File: internal/provider/kubernetes/httproute.go [hunk 13]
```diff
@@ -25,10 +24,10 @@ type httpRouteReconciler struct {
 
 // newHTTPRouteController creates the httproute controller from mgr. The controller will be pre-configured
 // to watch for HTTPRoute objects across all namespaces.
-func newHTTPRouteController(mgr manager.Manager, cfg *config.Server) error {
+func newHTTPRouteController(mgr manager.Manager, logger logr.Logger) error {
```
Comment (danehans) [inline internal/provider/kubernetes/httproute.go:27]: Same comment as https://github.com/envoyproxy/gateway/pull/197#discussion_r942693010.

File: internal/provider/kubernetes/gatewayclass.go [hunk 14]
```diff
@@ -30,26 +30,26 @@ type gatewayClassReconciler struct {
...
 	initializeOnce sync.Once
-	resourceTable  *ResourceTable
+	resources      *message.ProviderResources
 }
 
 // newGatewayClassController creates the gatewayclass controller. The controller
 // will be pre-configured to watch for cluster-scoped GatewayClass objects with
 // a controller field that matches name.
-func newGatewayClassController(mgr manager.Manager, cfg *config.Server, resourceTable *ResourceTable) error {
+func newGatewayClassController(name string, mgr manager.Manager, logger logr.Logger, resources *message.ProviderResources) error {
```
Comment (danehans) [inline internal/provider/kubernetes/gatewayclass.go:39]: Same comment as https://github.com/envoyproxy/gateway/pull/197#discussion_r942693010.

File: internal/provider/kubernetes/gateway.go (hunk 10 above)
Comment (danehans) [inline internal/provider/kubernetes/gateway.go:39]: Same comment as https://github.com/envoyproxy/gateway/pull/197#discussion_r942693010.

File: internal/provider/kubernetes/gatewayclass.go [hunk 15]
```diff
@@ -30,26 +30,26 @@ type gatewayClassReconciler struct {
...
 	if err := mgr.Add(uh); err != nil {
 		return fmt.Errorf("failed to add status update handler %v", err)
 	}
 
-	resourceTable.Initialized.Add(1)
+	resources.Initialized.Add(1)
 	r := &gatewayClassReconciler{
 		client:        cli,
-		controller:    gwapiv1b1.GatewayController(cfg.EnvoyGateway.Gateway.ControllerName),
+		controller:    gwapiv1b1.GatewayController(name),
```
Comment (danehans) [inline internal/provider/kubernetes/gatewayclass.go:49]: The value of `controller` should continue to be exposed to users to support running multiple Envoy Gateways in a cluster.

**Review by danehans — CHANGES_REQUESTED** [review]
@arkodg this is a great start! Other than a few nits, test coverage, and passing CI, this LGTM.

File: internal/xds/server/service/service.go [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+	controlplane_service_route_v3.RegisterRouteDiscoveryServiceServer(g, srv)
+	controlplane_service_runtime_v3.RegisterRuntimeDiscoveryServiceServer(g, srv)
+}
+
+func (s *Service) subscribeAndTranslate(ctx context.Context) {
+	// Subscribe to resources
+	xdsCh := s.XdsResources.Subscribe(ctx)
+	for ctx.Err() == nil {
+		// Receive subscribed resource notifications
+		<-xdsCh
```
Comment (LukeShu) [inline internal/xds/server/service/service.go:86]: If you only have 1 subscription to watch, you can just do

```
for _ = range s.XdsResources.Subscribe(ctx)  {
```
The subscription channel will get closed when the context gets canceled.

File: internal/message/types.go [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+	watchable.Map[string, *xdstypes.XdsResources]
+}
+
+func (x *XdsResources) Get() *xdstypes.XdsResources {
+	// Itreate and return the first element
+	for _, v := range x.LoadAll() {
+		return v
+	}
+	return nil
+}
```
Comment (LukeShu) [inline internal/message/types.go:76]: Do we expect to track multiple sets of resources, and this is a short-term hack? Or is this a hack around not being able to have a single "`watchable.Value`" without wrapping it in a map?

File: internal/message/types.go [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+
+// XdsResources message
+type XdsResources struct {
+	watchable.Map[string, *xdstypes.XdsResources]
+}
+
+func (x *XdsResources) Get() *xdstypes.XdsResources {
+	// Itreate and return the first element
```
Comment (LukeShu) [inline internal/message/types.go:71]: typo: Iterate

**Review by LukeShu — COMMENTED** [review]
Yeah, this is basically how I think `watchable` should be used. I left a suggestion to simplify the simple cases. If we think we'll have a lot of non-map uses, adding a `watchable.Value` would be good.

**Review by youngnick — COMMENTED** [review]
Took a really quick look over this and had one naming thing, sorry to be picky, but I think it could really bite us down the line if we don't fix it now.

I'll have more of a look at the rest later.

File: tools/make/lint.mk [hunk 19]
```diff
@@ -25,7 +25,7 @@ lint: lint.yamllint
 lint-deps: $(tools/yamllint)
 lint.yamllint: $(tools/yamllint)
 	@echo Running YAML linter ...
-	$(tools/yamllint) --config-file=tools/linter/yamllint/.yamllint changelogs/ internal/provider/kubernetes/config/ internal/provider/testdata/in/ internal/envoygateway/config/testdata/decoder/in/ internal/gatewayapi/testdata/
```
[the remark this answers is not in the document]
Comment (arkodg) [reply tools/make/lint.mk:28]: `xargs -L1 dirname` seems to work for me

File: internal/cmd/server.go [hunk 20]
```diff
@@ -1,13 +1,17 @@
...
 
 	"github.com/envoyproxy/gateway/api/config/v1alpha1"
 	"github.com/envoyproxy/gateway/internal/envoygateway/config"
-	"github.com/envoyproxy/gateway/internal/provider"
+	gatewayapisvc "github.com/envoyproxy/gateway/internal/gatewayapi/service"
+	infrasvc "github.com/envoyproxy/gateway/internal/infrastructure/service"
+	"github.com/envoyproxy/gateway/internal/message"
+	providersvc "github.com/envoyproxy/gateway/internal/provider/service"
+	xdsserversvc "github.com/envoyproxy/gateway/internal/xds/server/service"
+	xdstranslatorsvc "github.com/envoyproxy/gateway/internal/xds/translator/service"
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/cmd/server.go:14]: neutral here so renaming to `runner` as suggested by @youngnick and will adopt naming as `<pkgbase>/<pkgbase><subpkg>` as suggested by @LukeShu 
slightly off topic, @LukeShu what renaming scheme do you recommend for envoy libs such as https://github.com/envoyproxy/gateway/blob/969f91c89dccf573daed614b6b3871ba0c596198/internal/cmd/xdstest.go#L15

File: internal/provider/kubernetes/kubernetes.go [hunk 21]
```diff
@@ -53,20 +41,19 @@ func New(cfg *rest.Config, svr *config.Server, resourceTable *ResourceTable) (*P
...
-	if err := newGatewayClassController(mgr, svr, resourceTable); err != nil {
+	if err := newGatewayClassController(controllerName, mgr, logger, resources); err != nil {
 		return nil, fmt.Errorf("failed to create gatewayclass controller: %w", err)
 	}
-	if err := newGatewayController(mgr, svr, resourceTable); err != nil {
+	if err := newGatewayController(controllerName, mgr, logger, resources); err != nil {
 		return nil, fmt.Errorf("failed to create gateway controller: %w", err)
 	}
-	if err := newHTTPRouteController(mgr, svr); err != nil {
+	if err := newHTTPRouteController(mgr, logger); err != nil {
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/provider/kubernetes/kubernetes.go:50]: this ties to `gwapiv1b1.GatewayController(name)` / class controller, not the same of the sw controller/reconciler

File: internal/provider/kubernetes/gateway.go (hunk 10 above)
Comment (arkodg) [reply internal/provider/kubernetes/gateway.go:39]: I always prefer using function args to pass data around over context unless I absolutely need to ( and when the data is tied to a context rather than the package).
I can change the function signature and use a Config object instead to reduce the number of params passed to it

File: internal/cmd/server.go (hunk 1 above)
Comment (arkodg) [reply internal/cmd/server.go:90]: renamed to Runner with a New method

File: internal/message/types.go (hunk 17 above)
Comment (arkodg) [reply internal/message/types.go:76]: its the latter, wrapping it to return a single value embedded in a map (with a single key-value pair)

File: internal/gatewayapi/service/service.go [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+func (s *Service) Name() string {
+	return "gateway-api"
+}
+
+// Start starts the GatewayAPI service
+func (s *Service) Start(ctx context.Context) error {
+	s.Logger = s.Logger.WithValues("service", s.Name())
+	// Wait until provider resources have been initialized during startup
+	s.ProviderResources.Initialized.Wait()
+	go s.subscribeAndTranslate(ctx)
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/gatewayapi/service/service.go:29]: * used `ctx.Done()` within `Start` to wait for shutdown and then perform graceful shutdown steps needed within the `runner`, for eg, stopping the grpc server within the xds server runner
* Launching such go routines is not at all ideal and is a temporary solution until we implement something better using https://github.com/envoyproxy/gateway/issues/43
* for this specific routine, dont think any cleanup is needed during shutdown

File: internal/provider/kubernetes/gatewayclass.go (hunk 15 above)
Comment (arkodg) [reply internal/provider/kubernetes/gatewayclass.go:49]: not sure what you are referring to here

**Comment by youngnick** [comment]
`runner` is great, thanks for renaming. I won't LGTM because I haven't had a chance to check over how everything works, but my previous objection is addressed, so please don't let my not reviewing block this.

File: internal/gatewayapi/service/service.go (hunk 22 above)
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/gatewayapi/service/service.go]: 🙈 the ctx.Done() was living in the wrong method, moved it over to the goroutine

File: internal/provider/kubernetes/httproute.go (hunk 13 above)
Comment (arkodg) [reply internal/provider/kubernetes/httproute.go]: fixed

**Comment by arkodg** [comment]
this PR is ready to be reviewed now. I'd prefer not to increase the size of this PR anymore and address other issues in a follow up PR

Two issue I'd like to highlight
* the provider fails while starting with this error
```
no matches for kind "HTTPRoute" in version "[gateway.networking.k8s.io/v1beta1](http://gateway.networking.k8s.io/v1beta1)"
```
from[ here](https://github.com/envoyproxy/gateway/blob/2be093c0dc2a662739a060c131250aeb1fd915f0/internal/provider/kubernetes/httproute.go#L80)
this issue is not tied to this PR but is likely introduced by 
https://github.com/envoyproxy/gateway/pull/194 and is present on `main` too. I have commented the snippet causing the issue and this can be resolved in a follow up PR after reopening https://github.com/envoyproxy/gateway/issues/189
cc @chauhanshubham 
* looks like there is always a watchable subscription notification during init
I've circumvented it by skipping those messages
```
if msg == nil {
  continue
}
```
but im wondering if this is by design @LukeShu  ?

**Comment by arkodg** [comment]
looks like the 1st error highlighted in https://github.com/envoyproxy/gateway/pull/197#issuecomment-1220192374 is not an issue, reverted the code, and installed the right version of GAPI CRDs(0.5.0) which installed the v1beta1 version for HTTPRoute which is used internally in the code, thank you  @chauhanshubham for pointing this out 🙇

File: internal/provider/kubernetes/config/default/auth_proxy_patch.yaml [hunk 23]
```diff
@@ -4,7 +4,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: envoy-gateway
-  namespace: system
```
Comment (danehans) [inline internal/provider/kubernetes/config/default/auth_proxy_patch.yaml:7]: This change is unneeded due to https://github.com/envoyproxy/gateway/blob/main/internal/provider/kubernetes/config/default/kustomization.yaml#L1-L2

File: internal/provider/kubernetes/config/default/auth_proxy_patch.yaml [hunk 24]
```diff
@@ -27,8 +27,3 @@ spec:
           requests:
             cpu: 5m
             memory: 64Mi
-      - name: envoy-gateway
-        args:
-        - "--health-probe-bind-address=:8081"
-        - "--metrics-bind-address=127.0.0.1:8080"
-        - "--leader-elect"
```
Comment (danehans) [inline internal/provider/kubernetes/config/default/auth_proxy_patch.yaml:34]: Why are these changes required?

File: internal/provider/kubernetes/config/default/config_patch.yaml [hunk 25]
```diff
@@ -2,18 +2,18 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: envoy-gateway
-  namespace: system
+  namespace: envoy-gateway-system
```
Comment (danehans) [inline internal/provider/kubernetes/config/default/config_patch.yaml:5]: This change is unneeded due to https://github.com/envoyproxy/gateway/blob/main/internal/provider/kubernetes/config/default/kustomization.yaml#L1-L2

File: internal/provider/kubernetes/config/default/kustomization.yaml [hunk 26]
```diff
@@ -6,7 +6,7 @@ namespace: envoy-gateway-system
 # "wordpress" becomes "alices-wordpress".
 # Note that it should also match with the prefix (text before '-') of the namespace
 # field above.
-namePrefix: envoy-gateway-
+# namePrefix: envoy-gateway-
```
Comment (danehans) [inline internal/provider/kubernetes/config/default/kustomization.yaml:9]: See my other comments in reference to changing this. Not sure why this change is needed.

File: internal/provider/kubernetes/config/envoy-gateway/kustomization.yaml [hunk 27]
```diff
@@ -7,4 +7,4 @@ generatorOptions:
 configMapGenerator:
 - name: envoy-gateway-config
   files:
-  - envoy-gateway.yaml
+  - config.yaml
```
Comment (danehans) [inline internal/provider/kubernetes/config/envoy-gateway/kustomization.yaml:10]: We now have a config file mounted in a config directory. I thought the common pattern is the name of the config, e.g. envoy-gateway in the config directory.

File: tools/make/kube.mk [hunk 28]
```diff
@@ -1,11 +1,14 @@
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION ?= 1.24.1
+# GATEWAY_API_VERSION refers to the version of Gateway API CRDs.
+# For more details, see https://gateway-api.sigs.k8s.io/v1alpha2/guides/getting-started/#installing-gateway-api-crds-manually
+GATEWAY_API_VERSION ?= 0.5.0
```
Comment (danehans) [inline tools/make/kube.mk:5]: We should have a comment in go.mod that states this var should be updated when updating the module or:
```
GATEWAY_API_VERSION=$(go list -m -f '{{.Version}}' sigs.k8s.io/gateway-api)
```

**Review by danehans — COMMENTED** [review]
Please review my feedback. I'm not opposed to approving the PR with the suggested changes or filing issues to track the suggested changes.

File: internal/cmd/server.go [hunk 29]
```diff
@@ -48,19 +66,94 @@ func server() error {
...
 		}
 		// Set defaults for unset fields
 		eg.SetDefaults()
 		cfg.EnvoyGateway = eg
 	}
+	return cfg, nil
+}
+
+// setupServices starts all the services required for the Envoy Gateway to
+// fulfill its tasks.
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/cmd/server.go:79]: I reverted all the code based on @youngnick's suggestion, I'll rename this function to `setupRunners` make this more logical. This name is not final and would prefer if we can revisit this in the near future.

File: internal/gatewayapi/runner/runner.go [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+	// Wait until provider resources have been initialized during startup
+	r.ProviderResources.Initialized.Wait()
+	r.Logger.Info("done initializing provider resources")
+	// Subscribe to resources
+	gatewayClassesCh := r.ProviderResources.GatewayClasses.Subscribe(ctx)
+	gatewaysCh := r.ProviderResources.Gateways.Subscribe(ctx)
+	httpRoutesCh := r.ProviderResources.HTTPRoutes.Subscribe(ctx)
+
+	// Wait until provider resources have been initialized during startup
+	r.ProviderResources.Initialized.Wait()
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/gatewayapi/runner/runner.go:50]: good catch

File: internal/provider/kubernetes/config/default/auth_proxy_patch.yaml (hunk 24 above)
Comment (arkodg) [reply internal/provider/kubernetes/config/default/auth_proxy_patch.yaml:34]: these args were being appended to the `envoy-gateway` container, which I dont think is the intent

File: internal/provider/kubernetes/config/envoy-gateway/kustomization.yaml (hunk 27 above)
Comment (arkodg) [reply internal/provider/kubernetes/config/envoy-gateway/kustomization.yaml:10]: sure can rename config.yaml to envoy-gateway.yaml if thats the preference

File: tools/make/kube.mk (hunk 28 above)
Comment (arkodg) [reply tools/make/kube.mk:5]: nice trick, updating the PR

File: internal/provider/kubernetes/config/default/config_patch.yaml (hunk 25 above)
Comment (arkodg) [reply internal/provider/kubernetes/config/default/config_patch.yaml:5]: uncommenting the above line changes the name of the `envoy-gateway` pod to `envoy-gateway-envoy-gateway`, do we want that ?

File: internal/gatewayapi/runner/runner.go [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+		r.Logger.Info("received a notification")
+		// Load all resources required for translation
+		in.Gateways = r.ProviderResources.GetGateways()
+		in.HTTPRoutes = r.ProviderResources.GetHTTPRoutes()
+		gatewayClasses := r.ProviderResources.GetGatewayClasses()
+		// Fetch the first gateway class since there should be only 1
+		// gateway class linked to this controller
+		t := &gatewayapi.Translator{
+			GatewayClassName: v1beta1.ObjectName(gatewayClasses[0].GetName()),
+		}
```
[the remark this answers is not in the document]
Comment (arkodg) [reply internal/gatewayapi/runner/runner.go:68]: raised https://github.com/envoyproxy/gateway/issues/209 to help answer and edit this section based on the outcome of the discussion

**Comment by LukeShu** [comment]
Wait, is there no CI checking that `make generate` is up-to-date?

**Comment by arkodg** [comment]
> Wait, is there no CI checking that `make generate` is up-to-date?

not yet, raised https://github.com/envoyproxy/gateway/issues/213 to track it

**Review by LukeShu — APPROVED** [review]
